### PR TITLE
Fix model loading on big endian.

### PR DESCRIPTION
**Pull request by mahiuchun**
This solves the same problem as that of PR#150. Just want to explore some different approaches.

Tested on PPC64.

### CODE REVIEW

File: src/model.cpp [hunk 1]
```diff
@@ -39,12 +40,27 @@ using LibMatrix::vec2;
 using LibMatrix::vec3;
 using LibMatrix::uvec3;
 
-#define read_or_fail(file, dst, size) do { \
-    file.read(reinterpret_cast<char *>((dst)), (size)); \
-    if (file.gcount() < (std::streamsize)(size)) { \
+static inline void read_from_file(std::istream& file, void *dst, size_t nmemb, size_t size) {
+  file.read(reinterpret_cast<char *>(dst), nmemb * size);
```
Comment (afrantzis) [inline src/model.cpp:44]: Indent size is 4 (not 2). Similarly below.

**Review by afrantzis — CHANGES_REQUESTED** [review]
Hi! A small code formatting nit (see inline comment), but otherwise looks good.

File: src/model.cpp (hunk 1 above)
Comment (mahiuchun) [reply src/model.cpp]: Done

**Comment by afrantzis** [comment]
Rebased and merged in https://github.com/glmark2/glmark2/commit/2ddc86ff65aa93abc321f98996c2657e9fd2d712 . Thanks!

**Pull request closed without merging**